Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp

```diff
@@ -967,7 +994,7 @@ RefPtr<WebCore::WebGPU::GPU> WebChromeClient::createGPUForWebGPU() const
 RefPtr<WebCore::ShapeDetection::BarcodeDetector> WebChromeClient::createBarcodeDetector(const WebCore::ShapeDetection::BarcodeDetectorOptions& barcodeDetectorOptions) const
 {
 #if ENABLE(GPU_PROCESS)
-    auto& remoteRenderingBackendProxy = m_page.ensureRemoteRenderingBackendProxy();
+    auto& remoteRenderingBackendProxy = protectedPage()->ensureRemoteRenderingBackendProxy();
```

Comment:
This doesn't seem right. Why is it safe to hold the reference from this function without protecting the page in a local variable?